src/widget/weffectparameternamebase.cpp

```diff
@@ -35,6 +35,14 @@ void WEffectParameterNameBase::setEffectParameterSlot(
                 &EffectParameterSlotBase::updated,
                 this,
                 &WEffectParameterNameBase::parameterUpdated);
+        if (qobject_cast<EffectKnobParameterSlot*>(m_pParameterSlot.data())) {
```

Comment:
Why is this preferred over
`if (m_pParameterSlot->parameterType() == EffectParameterType::Knob)`
and use data() twice?

Reply:
data() has no cost because it is an inline function just returning the pointer. 
https://github.com/qt/qtbase/blob/067b53864112c084587fa9a507eb4bde3d50a6e1/src/corelib/tools/qsharedpointer_impl.h#L271

I have decided to use the qobject_cast, because it makes it obvious that the result will not change during the lifetime of 
m_pParameterSlot and the connection in this place is sufficient. Does this work for you?

Reply:
Sure, I was just curious.